### ISSUE TITLE
Incorrect size of cell with multiline text on PDF export

### ISSUE THREAD

**Issue by vsisko** [hhurz/tableExport.jquery.plugin#42]
Hello. When I try to export table like this:

``` html
<table id="tbl">
    <thead>
        <tr>
            <th>Title</th>
        </tr>
    </thead>
    <tbody>
        <tr>
            <td>
                Very long long long long long long long long long long long long long
                long long long long long long long long long long long long long long
                long long long long long long long long long long long long long Message
            </td>
        </tr>
        <tr>
            <td>
                Second row
            </td>
        </tr>
    </tbody>
</table>
```

to PDF format with properties:

```
{type:'pdf',
  jspdf: {
    autotable: {
      styles: {overflow: 'linebreak',
        fontSize: 26,
        rowHeight: 36,
        fillColor: 155},
      headerStyles: {rowHeight: 28,
        fontSize: 38}
   }
  }
}
```

height of cell is too small for content showing. Incorrect size cut bottom part of text.:
![html to pdf cell height](https://cloud.githubusercontent.com/assets/5163151/10686515/c4a2433c-7986-11e5-9bb7-5b5cfdf85789.png)

Padding and margin sizes do not change view of generated file.

**Comment by hhurz** [comment]
Think I solved that issue. But the solution is more like a workaround cause the underlying plugins (jsPDF and/or jsPDF-Autotable) do not calculate the line height properly.

**Comment by vsisko** [comment]
Row height look better now, but after cutting of a cell on a page break only a title is shown on second page:

``` html
<table id="tbl">
    <thead>
        <tr>
            <th>Title</th>
        </tr>
    </thead>
    <tbody>
    <tr><td>row 1<td/><tr/> <tr><td>row 2<td/><tr/> <tr><td>row 3<td/><tr/> <tr><td>row 4<td/><tr/> <tr><td>row 5<td/><tr/>
    <tr><td>row 6<td/><tr/> <tr><td>row 7<td/><tr/> <tr><td>row 8<td/><tr/> <tr><td>row 9<td/><tr/> <tr><td>row 10<td/><tr/>
    <tr><td>row 11<td/><tr/> <tr><td>row 12<td/><tr/> <tr><td>row 13<td/><tr/>
    <tr>
        <td style="margin: 5px">
            Very long long long long long long long long long long long long long
            long long long long long long long long long long long long long long
            long long long long long long long long long long long long long Message
        </td>
    </tr>
    <tr><td>Next page row 1<td/><tr/>
    <tr><td>Next page row 2<td/><tr/> 
    </tbody>
</table>
```

On exported PDF with properties:

```
{type:'pdf',
    jspdf: {
        autotable: {
            styles: {overflow: 'linebreak',
                fontSize: 26,
                rowHeight: 36,
                fillColor: 155},
            headerStyles: {rowHeight: 28,
                fontSize: 38}
        }
    }
}
```

![html page break empty new page](https://cloud.githubusercontent.com/assets/5163151/10723150/1b6907d4-7be2-11e5-88b3-a9bc4c3139ed.png)

**Comment by vsisko** [comment]
I set valign property to **top** on previous version of library and see sumular paddings in top and bottom of cells. Look like top padding calculated incorrectly for **middle** and **bottom** value of valign property:

```
{type:'pdf',
    jspdf: {
        autotable: {
            styles: {overflow: 'linebreak',
                fontSize: 26,
                rowHeight: 36,
                fillColor: 155,
                valign: 'top'}
        }
    }
}
```

![html to pdf valign](https://cloud.githubusercontent.com/assets/5163151/10723998/d9e9765a-7bea-11e5-807d-9c8f88008c39.png)

**Comment by hhurz** [comment]
Thanks for testing again and leading me to the valign option of jsPDF-AutoTable. I "overlayed" its autoTableText function and added some "magic" text position corrections depending on the valign settings. Thats a workaround again as long as the dimension of text containing linebreaks isn't calcualted correctly in the underlying plugins.

**Comment by hhurz** [comment]
In v1.2.0 of tableExport I added a new fix for jsPDF AutoTables row height calculation. Things should be better now...

**Comment by vsisko** [comment]
Top and bottom cell paddings is different for different value of valign property.
In example 42 only one cell with long text and in that variant top padding should be equal to bottom padding for any value of valign property. 

I create a new example with two cell in row. The first cell have multiline message and should stretch cell by content with similar top and bottom padding. The second cell should be aligned accordance to valign property:

``` html
<table id="tbl">
    <thead>
        <tr>
            <th>Long value</th>
            <th>Short value</th>
        </tr>
    </thead>
    <tbody>
        <tr>
            <td>
                Very long long long long long long long long long long long long Message
            </td>
            <td>Aligned cell</td>
        </tr>
        <tr>
            <td>Second row cell 1</td>
            <td>Second row cell 2</td>
        </tr>
    </tbody>
</table>
```

Exported to PDF with properties:

```
{type:'pdf',
  jspdf: {
    autotable: {
      styles: {overflow: 'linebreak',
        fontSize: 26,
        rowHeight: 36,
        fillColor: 155,
        valign: 'middle'},
      headerStyles: {rowHeight: 28,
        fontSize: 38}
   }
  }
}
```

![pdf 2 cols align example](https://cloud.githubusercontent.com/assets/5163151/10837472/02fa58a6-7ee4-11e5-9430-5fcb79025ca3.png)

**Comment by hhurz** [comment]
With export properties from above where valign is set to 'middle',  you are right with the multiline text. It is not properly vertical aligned. But the short value cell text aligned correctly. 

Text positioning of both cells works for valign: 'top' 

![vtop](https://cloud.githubusercontent.com/assets/4164372/10841843/115b0ea0-7eed-11e5-830f-bd8fa508704b.png)

When using valign: 'bottom' the position of the multiline text is wrong, but the single line text fits

![vbottom](https://cloud.githubusercontent.com/assets/4164372/10841900/b2db3228-7eed-11e5-8b4c-2f7337a89b84.png)

I'll have a closer look on that.

**Comment by hhurz** [comment]
Next try: issue should be fixed in v1.2.2
